### PR TITLE
Added cm-300 board files & deleted unnecessary files.

### PR DESCRIPTION
**Pull request by robotisApp**
- Merge kigateG's repo.
- Deleted unnecessary files.
- Modified .gitignore.

- Added robotis-cm-300 lib to support CM-300 board.
- Updated platform files for core---nrf52.

### CODE REVIEW

**Comment by pelikhan** [comment]
Do not use travis, it will not work for private repoes.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+language: node_js
```
Comment (pelikhan) [inline .travis.yml]: travis will not work for priate repoes. use github actions instead

**Comment by kigateG** [comment]
Hi. 
How can I fix the 2 errors. hex file & compiler test failed.
I think I saw the message in the following cases.

1.
["compileServiceVariant": "nrf52840"] in [pxt-robotis/libs/robotis-cm-300-01/pxt.json].
[pxt-robotis/libs/robotis-cm-300-01-blocks] has cpp files.
=> no error

2.
["compileServiceVariant": "cm300"] in [pxt-robotis/libs/robotis-cm-300-01/pxt.json].
[pxt-robotis/libs/robotis-cm-300-01-blocks] has cpp files.
=> error

3.
["compileServiceVariant": "cm300"] in [pxt-robotis/libs/robotis-cm-300-01/pxt.json].
[pxt-robotis/libs/robotis-cm-300-01-blocks] has NO cpp files.
=> no error

BUT, now....
Nothing special has been modified, but case2. shows No error.

**Comment by pelikhan** [comment]
There is still a ton of useless libs in there like "esp32". You need to clean up and remove all the unecessary libs first then add them back later. When do you plan to get this PR ready?

Note: i don't like to review massive PRs so use 1 PR per review.

**Comment by kigateG** [comment]
Hello. I found a similar issue. https://github.com/Microsoft/pxt-maker/issues/73

It works fine in local PC with command "pxt serve --localbuild".
But Bad HTTP status code: 412 on Github.

What should I do?

**Comment by pelikhan** [comment]
This PR is too large to review. Please merge it and identify issues in separate smaller chunks. I don't have the time to review 800+ files PRs.

**Comment by kigateG** [comment]
> This PR is too large to review. Please merge it and identify issues in separate smaller chunks. I don't have the time to review 800+ files PRs.

Thank you.
First, I will delete the unnecessary code and leave a comment.

**Pull request closed without merging**